modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -482,4 +513,61 @@ private void setOperationIsAsync() {
             additionalProperties.put(OPERATION_IS_ASYNC, operationIsAsync);
         }
     }
+
+    private void setIsFramework() {
+        if (aspnetCoreVersion.getOptValue().startsWith("3.")) {
+            // default, do nothing
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so changing  to use frameworkReference instead of packageReference ");
```

Comment:
This sets `isFramework` which may indirectly affect `frameworkReference`, but the log message should describe what's being done by the generator rather than the template. For example, someone could create a custom template, and could find it confusing if they've hard-coded frameworkReference or something else, and yet the generator says it's "changing to use frameworkReference".

Reply:
Changed the variable to useFrameworkReference to make things explicitly clear. Its one of the breaking differences between 2.x and 3.x - for the framework the reference has to be by FrameworkReference. 

I think if people hard code stuff that contradicts the setting there might be little we can do about it.